rust/core-lib/src/tabs.rs

```diff
@@ -491,6 +496,27 @@ impl<W: Write + Send + 'static> Documents<W> {
                 self.plugins.stop_plugin(&view_id, &plugin_name);
                 None
             }
+            PluginRpc  { view_id, receiver, mut rpc } => {
+                // inserting view_id into params here because the plugin needs
+                // to know where this command originates. This feels a bit
+                // clumbsy though. At the very least we should use some obscure
+                // key so we don't collide with any possible user-supplied keys?
+                // Or we could validate keys on plugin load or something.
+                assert!(rpc.params_ref().is_object(), "params must be an object");
+                rpc.params_ref_mut().as_object_mut()
+                    .map(|p| p.insert("view_id".to_owned(), json!(view_id)));
+                match rpc {
+                    PlaceholderRpc::Notification { ref method, ref params } => {
+                        self.plugins.dispatch_notification(&view_id, &receiver,
+                                                           method, params);
+                        None
+                    }
+                    PlaceholderRpc::Request { ref method, ref params } => {
```

Comment:
Do you have any examples of a situation where a frontend would want a response from a plugin?

In Sublime and Emacs at least there is no concept of a return value for a command, they're asynchronous things that you just fire and forget, and it doesn't seem to hamstring them.